### PR TITLE
✨ Feat/take leverage

### PR DESCRIPTION
**Pull request by Thomas-Smets**
Together with https://github.com/arcadia-finance/arcadia-lending/pull/39

### CODE REVIEW

File: src/Vault.sol [hunk 1]
```diff
@@ -199,7 +199,6 @@ contract Vault {
      */
     function openTrustedMarginAccount(address protocol) public onlyOwner {
         require(!isTrustedCreditorSet, "V_OMA: ALREADY SET");
-        //ToDo: Check in Factory/Mainregistry if protocol is indeed trusted?
 
         (bool success, address baseCurrency_, address liquidator_) =
             ITrustedCreditor(protocol).openMarginAccount(vaultVersion);
```
Comment (j-vp) [inline src/Vault.sol:204]: make sure interface is a view function, otherwise this is an open door

File: src/Vault.sol [hunk 2]
```diff
@@ -224,7 +222,6 @@ contract Vault {
         require(ITrustedCreditor(trustedCreditor).getOpenPosition(address(this)) == 0, "V_CMA: NON-ZERO OPEN POSITION");
```
Comment (j-vp) [inline src/Vault.sol:222]: same, doublecheck interface view

File: src/test/9_Liquidator.t.sol [hunk 3]
```diff
@@ -302,6 +302,7 @@ contract LiquidatorTest is DeployArcadiaVaults {
     {
         vm.assume(claimAmount <= openClaim);
         vm.assume(claimAmount <= liquidatorBalance);
+        vm.assume(claimer != liquidityProvider);
```
Comment (j-vp) [inline src/test/9_Liquidator.t.sol:305]: Is this just for simpler balance's in the assertEq's?

File: src/Vault.sol (hunk 1 above)
Comment (Thomas-Smets) [reply src/Vault.sol:204]: Good catch, should make a find the bug out of this for Twitter clout

File: src/test/9_Liquidator.t.sol (hunk 3 above)
Comment (Thomas-Smets) [reply src/test/9_Liquidator.t.sol:305]: Yes, otherwise assertEq balance fails

**Review by j-vp — APPROVED** [review]
gtg

**Pull request merged**